general/v0.1/registration.md

```diff
@@ -27,112 +27,101 @@ administrator. This MAY include but is not limited to the following:
 * Email address and/or other contact data of the adminstrator
 * A password of other means of authentication, so the administrator can
   sign-in again later
-* URL of the fediverse server(s) to be registered
 * Acceptance of terms of service, data processing agreement, and/or privacy policy
 
-A successful registration results in the FASP creating an OAuth 2.0
-application for the fediverse server, including a client identifier (ID) and
-secret. FASP MUST grant the application all the scopes defined here and in
-the FASP specifications of the capabilities the FASP advertises
-to the fediverse server (see section [04: Provider Info](provider_info.md)).
-FASP specifications MAY define exceptions to this rule.
+As part of the registration process the fediverse server administrator
+MUST provide the URL of their server. The FASP MUST use this URL to
+discover the base URL for FASP interaction using the
+`.well-known/nodeinfo` mechanism as described in [protocol basics](protocol_basics.md).
 
-After registration, FASP MUST present the fediverse server
-administrator a registration token that can be copied into the fediverse server configuration.
+A successful registration results in the FASP creating an Ed25519
+keypair and a client identifier (ID) for the fediverse server.
 
-The registration token is a JSON Web Token (JWT) that MUST include the
-registered claims `iss` and `sub` and the private claim `sec`. It MAY
-contain the registered claim `exp`.
+After registration, FASP MUST make an HTTP `POST` request to the
+fediverse server's `/registration` endpoint.
 
-These claims MUST contain the following:
+The payload of that request is a JSON object with the following keys and
+values:
 
-* `iss`: The base URI for API requests to the FASP.
-* `sub`: The OAuth 2.0 client identifier
-* `sec`: The OAuth 2.0 client secret
-* `exp`: An optional expiration time
+* `name`: The name of the FASP - this MUST have been presented to the
+  administrator during registration to make it recognizable.
+* `baseUrl`: The base URL of the FASP
+* `clientId`: The client ID
```

Comment:
We could drop `clientId` here and instead have it returned by the registration endpoint, this would likely reduce the chances of collisions and be easier to mandate on the Fediverse Server software side that the `clientId` must be a secure random base64url-encoded value.

So all you'd send in registration is:

```
{
  "name": "Example FASP",
  "url": "https://fasp.example.com",
  "publicKey": "..."
}
```

And then you'd get back the `clientId` for the registration.
```

Reply:
(I guess `"url"` isn't necessary in the Fediverse Server software response, since you've already discovered it via `nodeinfo`, but it would give a nice way to correlate the two)

Reply:
Oh, and I missed `redirectUri` here, though it may be better called something like `registrationCompletionUrl` ?

Reply:
"client_id" is a remnant from the OAuth proposal. There it might have made limited sense because it is an established term.

But I see that using this for both sides has more issues than just feeling out of place, so I renamed it `faspId` and `serverId` to make it really clear where it belongs to.

The FASP generates the `serverId`, because it needs it to be unique in its own DB. And the server the generates the `faspId` for the same reason, just the other way around.